docs/architecture/psa-thread-safety.md

```diff
@@ -1,10 +1,78 @@
-Thread safety of the PSA key store
+Thread safety of the PSA subsystem
 ==================================
 
-Analysis of the behavior of the PSA key store as of Mbed TLS 9202ba37b19d3ea25c8451fd8597fce69eaa6867.
+## Requirements
+
+### Backward compatibility requirement
+
+Code that is currently working must keep working. There can be an exception for code that uses features that are advertised as experimental; for example, it would be annoying but ok to add extra requirements for drivers.
+
+In particular, if you build with `MBEDTLS_PSA_CRYPTO_C` and `MBEDTLS_THREADING_C` and you either protect all PSA calls with a mutex, or only ever call PSA functions from a single thread, your application works.
```

Comment:
I'm not sure if we should remove the bit about `MBEDTLS_THREADING_C`. Regardless of it the rest of the sentence stands true.

Reply:
Good point, I was mixing up two slightly different ideas (if it compiled, it still compiles; if it worked, it still works). I've tried to clarify.